media_utils/eme_puppeteer.py

```diff
@@ -0,0 +1,138 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from time import sleep
+import re
+from re import compile
+
+from marionette_driver import By, expected, Wait
+from marionette_driver.errors import TimeoutException, NoSuchElementException
+
+# Adapted from https://github.com/gavinsharp/aboutmedia/blob/master/chrome/content/aboutmedia.xhtml
+_debug_script = "var mainWindow = window.QueryInterface(Components.interfaces.nsIInterfaceRequestor)\
+                       .getInterface(Components.interfaces.nsIWebNavigation)\
+                       .QueryInterface(Components.interfaces.nsIDocShellTreeItem)\
+                       .rootTreeItem\
+                       .QueryInterface(Components.interfaces.nsIInterfaceRequestor)\
+                       .getInterface(Components.interfaces.nsIDOMWindow);\
+var tabbrowser = mainWindow.gBrowser;\
+for (var i=0; i < tabbrowser.browsers.length; ++i) {\
+  var b = tabbrowser.getBrowserAtIndex(i);\
+  var media = b.contentDocumentAsCPOW.getElementsByTagName('video');\
+  for (var j=0; j < media.length; ++j) {\
+     var ms = media[j].mozMediaSourceObject;\
+     if (ms) {\
+       debugLines = ms.mozDebugReaderData.split(\"\\n\");\
+       return debugLines;\
+     }\
+  }\
+}"
+
+def playback_started(eme):
+    try:
+        current_time = eme.player_current_time
+        return current_time > 0
+    except BaseException as e:
```

Comment:
Do you mean to catch `Exception` instead? `BaseException` will catch stuff like `KeyboardInterrupt` and `SystemExit`.